vm/starnix/starnix.go

```diff
@@ -0,0 +1,374 @@
+// Copyright 2023 syzkaller project authors. All rights reserved.
+// Use of this source code is governed by Apache 2 LICENSE that can be found in the LICENSE file.
+
+package starnix
+
+import (
+	"fmt"
+	"io"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"strings"
+	"time"
+
+	"github.com/google/syzkaller/pkg/config"
+	"github.com/google/syzkaller/pkg/log"
+	"github.com/google/syzkaller/pkg/osutil"
+	"github.com/google/syzkaller/pkg/report"
+	"github.com/google/syzkaller/sys/targets"
+	"github.com/google/syzkaller/vm/vmimpl"
+)
+
+func init() {
+	var _ vmimpl.Infoer = (*instance)(nil)
+	vmimpl.Register("starnix", ctor, true)
+}
+
+type Config struct {
+	// Number of VMs to run in parallel (1 by default).
+	Count int `json:"count"`
+	// Location of Fuchsia checkout, for now we need this to run ffx commands.
+	Fuchsia string `json:"fuchsia"`
+}
+
+type Pool struct {
+	count   int
+	env     *vmimpl.Env
+	cfg     *Config
+	version string
+}
+
+type instance struct {
+	fuchsiaDirectory string
+	name             string
+	index            int
+	cfg              *Config
+	version          string
+	args             []string
+	debug            bool
+	workdir          string
+	timeouts         targets.Timeouts
+	port             int
+	rpipe            io.ReadCloser
+	wpipe            io.WriteCloser
+	fuchsiaVmStarted bool
+	fuchsiaLogs      *exec.Cmd
+	adb              *exec.Cmd
+	executor         string
+	merger           *vmimpl.OutputMerger
+	diagnose         chan bool
+}
+
+func getFuchsiaVersion(fuchsiaDirectory string) ([]byte, error) {
+	cmd := osutil.Command("ffx", "version")
```

Comment:
Where does `ffx` binary come from? Do you just happen to have some in PATH? Is it in the fuchsia checkout? If so, we need full path here.